geopandas/_block.py

```diff
@@ -0,0 +1,347 @@
+        from shapely.geometry.base import geom_factory
+        return geom_factory
+
+    # @property
+    # def _na_value(self):
+    #     return None
+    #
+    # @property
+    # def fill_value(self):
+    #     return tslib.iNaT
+
+    # TODO
+    # def copy(self, deep=True, mgr=None):
+    #     """ copy constructor """
+    #     values = self.values
+    #     if deep:
+    #         values = values.copy(deep=True)
+    #     return self.make_block_same_class(values)
+
+    def external_values(self):
+        """ we internally represent the data as a DatetimeIndex, but for
+        external compat with ndarray, export as a ndarray of Timestamps
+        """
+        #return np.asarray(self.values)
+        print("I am densified (external_values, {} elements)".format(len(self)))
+        return self.values.to_dense()
+
+    def formatting_values(self, dtype=None):
+        """ return an internal format, currently just the ndarray
+        this should be the pure internal API format
+        """
+        return self.to_dense()
+
+    def to_dense(self):
+        print("I am densified ({} elements)".format(len(self)))
+        return self.values.to_dense().view()
+
+    def _getitem(self, key):
+        values = self.values[key]
+        return GeometryBlock(values, placement=slice(0, len(values), 1),
+                             ndim=1)
+
+    # TODO is this needed?
+    # def get_values(self, dtype=None):
+    #     """
+    #     return object dtype as boxed values, as shapely objects
+    #     """
+    #     if is_object_dtype(dtype):
+    #         return lib.map_infer(self.values.ravel(),
+    #                              self._box_func).reshape(self.values.shape)
+    #     return self.values
+
+    def to_native_types(self, slicer=None, na_rep=None, date_format=None,
+                        quoting=None, **kwargs):
+        """ convert to our native types format, slicing if desired """
+
+        values = self.values
+        if slicer is not None:
+            values = values[slicer]
+
+        from geopandas.vectorized import to_shapely
+        values = to_shapely(values.data)
+
+        return np.atleast_2d(values)
+
+    # TODO needed for what?
+    def _can_hold_element(self, element):
+        # if is_list_like(element):
+        #     element = np.array(element)
+        #     return element.dtype == _NS_DTYPE or element.dtype == np.int64
+        from shapely.geometry.base import BaseGeometry
+        return isinstance(element, BaseGeometry)
+
+    def _slice(self, slicer):
+        """ return a slice of my values """
+        print("I am sliced")
+        if isinstance(slicer, tuple):
+            col, loc = slicer
+            if not is_null_slice(col) and col != 0:
+                raise IndexError("{0} only contains one item".format(self))
+            return self.values[loc]
+        return self.values[slicer]
+
+    def take_nd(self, indexer, axis=0, new_mgr_locs=None, fill_tuple=None):
```

Comment:
Yes, but I don't think we need a full `take_nd`, as a GeometryBlock (as opposed to other blocks) is always 1D. So that aspect is handled here, and we can just have a VectorizedGeometry.take that works similar as numpy's take. 
I will update the function here to use that